src/doublyLinkedList.ts

```diff
@@ -0,0 +1,83 @@
+// found this here - https://www.cnblogs.com/answer1215/p/7623103.html
+
+/**
+ * Linked list node
+ */
+export interface DoublyLinkedListNode<T> {
+	value: T;
+	next?: DoublyLinkedListNode<T>;
+	prev?: DoublyLinkedListNode<T>;
+}
+
+/**
+ * Linked list for items of type T
+ */
+export class DoublyLinkedList<T> {
+	public head?: DoublyLinkedListNode<T> = undefined;
+	public tail?: DoublyLinkedListNode<T> = undefined;
+
+	/**
+	* Adds an item in O(1)
+	**/
+	add(value: T) {
```

Comment:
Suggestion: Given that you are using `pop` as the removal function name I would suggest to name the add function `push` or you make it the other way round and rename the `pop` to `remove` or similar.